src/parse.js

```diff
@@ -25,7 +25,8 @@ var base10Re    = /^[1-9][0-9]*$/,
     numberRe    = /^(?![eE])[0-9]*(?:\.[0-9]*)?(?:[eE][+-]?[0-9]+)?$/,
     nameRe      = /^[a-zA-Z_][a-zA-Z_0-9]*$/,
     typeRefRe   = /^(?:\.?[a-zA-Z_][a-zA-Z_0-9]*)(?:\.[a-zA-Z_][a-zA-Z_0-9]*)*$/,
-    fqTypeRefRe = /^(?:\.[a-zA-Z_][a-zA-Z_0-9]*)+$/;
+    fqTypeRefRe = /^(?:\.[a-zA-Z_][a-zA-Z_0-9]*)+$/,
+    featuresRefRe = /features\.([a-zA-Z_]*)/;
```

Comment:
features can also have extensions (e.g. features.(pb.cpp).legacy_enum).  Do we need to special-case feature parsing like this though?  They're implemented as just options with special meaning, so the parser shouldn't really need to treat them differently (until it does feature resolution)

Reply:
I *think* so, because there's slightly different syntax for features than for other options, and because parseOption is I think the final layer before feature resolution (as it is for option resolution as well). If we don't do this special regex-matching, we'd basically be looking for open parentheses to determine whether it's a custom option, which we don't have for feature-matching. I'll expand the regex to include special options.

Reply:
the syntax for features is exactly the same as that for message-typed options.  There's nothing special about them except in how we use them